dependency_cache.go

```diff
@@ -48,16 +48,31 @@ type DependencyCache struct {
 
 	// UserAgent is the User-Agent string to use with requests.
 	UserAgent string
+
+	// DependencyMap optionally provides override URIs for BuildpackDependencies
+	Mappings []DependencyMapping
 }
 
 // NewDependencyCache creates a new instance setting the default cache path (<BUILDPACK_PATH>/dependencies) and user
 // agent (<BUILDPACK_ID>/<BUILDPACK_VERSION>).
-func NewDependencyCache(buildpack libcnb.Buildpack) DependencyCache {
+// Mappings will be read from <PLATFORM_DIR>/dependencies/mappings.toml
+func NewDependencyCache(context libcnb.BuildContext) (DependencyCache, error) {
+	mappingsFile, err := ReadMappingsFile(DefaultMappingsFilePath(context.Platform.Path))
+	if err != nil {
+		return DependencyCache{}, err
+	}
+	var mappings []DependencyMapping
+	for _, bpm := range mappingsFile.BuildpackMappings {
+		if bpm.BuildpackID == context.Buildpack.Info.ID {
+			mappings = bpm.Mappings
```

Comment:
There should also be a log message (not sure of how prominent we want it to be) that mentions that mappings are going to be used for this buildpack.

Reply:
Omitting this for now because the Logger is not injected into the constructor